### PR TITLE
Replace custom ca-certs fetching impl with mirage lib

### PR DESCRIPTION
**Pull request by aronerben**

### CODE REVIEW

File: lib/letters.ml [hunk 1]
```diff
@@ -176,6 +176,8 @@ let ca_path_peer_verifier path =
   Lwt.return (X509.Authenticator.chain_of_trust ~time:now certs)
 ;;
 
+let cert = ref None
```
Comment (aronerben) [inline lib/letters.ml]: I wanted to memoize the CA certs fetching, else it is done for each mail send, what do you think? This is under the assumption that the CA certs location does not change while the `letters` process is running.

**Comment by aronerben** [comment]
FYI: There is a compile error in this branch. (Because there was one in master already, not caused by these changes)

File: lib/letters.ml (hunk 1 above)
Comment (joseferben) [reply lib/letters.ml]: This needs to be documented and made very explicit imo but otherwise fine by me.

Comment (aronerben) [reply lib/letters.ml]: I added a comment to the `ref` and a phrase in the README.

**Review by joseferben — APPROVED** [review]
Looks good thanks!

**Pull request merged**